spec/integration/indirector/direct_file_server_spec.rb

```diff
@@ -2,6 +2,15 @@
 require 'matchers/include'
 
 require 'puppet/indirector/file_content/file'
+require 'puppet/indirector/file_metadata/file'
+
+def make_uri(path)
+  if Puppet.features.microsoft_windows?
+    "file:///#{path}"
+  else
+    "file://host#{path}"
+  end
```

Comment:
I'm confused as to why the old version of the tests specify `file://host`. The unit tests for direct_file_server use `Puppet::Util.path_to_uri` to generate a URI, and it doesn't contain an authority component on either windows or posix:

<pre>
irb(main):003:0> Puppet::Util.path_to_uri('/foo').to_s
=> "file:/foo"
</pre>

<pre>
irb(main):018:0* Puppet::Util.path_to_uri('C:/foo').to_s
=> "file:/C:/foo"
</pre>

I added the `path_to_uri` method in 0ce60a569604105e5e7ef47431d239bcd0d46b27, and what I found is that `file://<host>/C:/foo` is used to express UNC paths on windows (even though `C:/foo` is local). In order to refer to a local file, it is better to use either `file:/C:/foo` or `file:///C:/foo`, where the former is allowed because the authority component is optional (https://tools.ietf.org/html/rfc3986#section-3.2)

But I'm wondering why windows and posix aren't consistent about the authority component. Is the test modeling the current behavior (posix agents include it, but windows ones don't)? I'm guessing the current tests don't match reality and that we should remove the authority component in the posix case. If so, we can just call `path_to_uri` instead.